### PR TITLE
Fix #7867: Improve documentation of Diagnostic.Create properties argument

### LINKED ISSUE #7867: Improve documentation of Diagnostic.Create properties argument
Some overloads of `Diagnostic.Create` have a `properties` dictionary argument, but the XML doc comment doesn't explain what it's for (why you'd want to use it). I was just reviewing an analyzer/fixer pair where the analyzer sets a property on the diagnostics it creates, and the fixer takes different actions depending on the value of the property. If that's what the `properties` dictionary is in fact for, it would help if the documentation explained that.

### PR DESCRIPTION
**Pull request by lgolding**
@dotnet/roslyn-analysis

### CODE REVIEW

File: src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs [hunk 1]
```diff
@@ -44,7 +44,11 @@ public abstract partial class Diagnostic : IEquatable<Diagnostic>, IFormattable
         /// </summary>
         /// <param name="descriptor">A <see cref="DiagnosticDescriptor"/> describing the diagnostic.</param>
         /// <param name="location">An optional primary location of the diagnostic. If null, <see cref="Location"/> will return <see cref="Location.None"/>.</param>
-        /// <param name="properties">An optional set of properties of the diagnostic. If null, <see cref="Properties"/> will return <see cref="ImmutableDictionary{TKey, TValue}.Empty"/>.</param>
+        /// <param name="properties">
+        /// An optional set of name-value pairs by means of which the the analyzer that creates the diagnostic
```
Comment (srivatsn) [inline src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs]: typo: the the (copy\pasted below as well)

**Comment by srivatsn** [comment]
:+1:

File: src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs (hunk 1 above)
Comment (lgolding) [inline src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs]: Oh boy, I had to stare for a long time before I saw "the the". Nice catch :-) Will fix.

**Comment by tannergooding** [comment]
test prtest/win/dbg/eta please

**Comment by lgolding** [comment]
test prtest/win/dbg/eta please

**Comment by lgolding** [comment]
@dotnet-bot test prtest/win/dbg/eta please

**Pull request merged**